### PR TITLE
Implement @at in JS paths

### PR DESCRIPTION
**Pull request by kalman**

### CODE REVIEW

**Comment by kalman** [comment]
CC @aboodman 
Fixes https://github.com/attic-labs/noms/issues/2989

File: js/noms/src/path.js [hunk 1]
```diff
@@ -397,11 +407,9 @@ export class HashIndexPath extends KeyIndexable {
       getCurrentValue = cur => cur.getCurrent();
     } else if (value instanceof Map) {
       seq = value.sequence;
-      if (this.intoKey) {
-        getCurrentValue = cur => cur.getCurrent()[0]; // key
-      } else {
-        getCurrentValue = cur => cur.getCurrent()[1]; // value
-      }
+      getCurrentValue = this.intoKey ?
```
Comment (arv) [inline js/noms/src/path.js]:
```js
getCurrentValue = cur => cur.getCurrent()[this.intoKey ? KEY : VALUE];
```


or

```js
const prop = this.intoKey ? KEY : VALUE;
getCurrentValue = cur => cur.getCurrent()[prop];
```

Comment (kalman) [inline js/noms/src/path.js]: I like.

**Comment by kalman** [comment]
Oops.

**Pull request merged**